### ISSUE TITLE
`@typedef` doesn't give error for bad or missing typename

### ISSUE THREAD

**Issue by sandersn** [Microsoft/TypeScript#26776]
```js
/** @typedef {number} 3 */
/** @typedef {number */
```

**Expected behavior:**

Errors on both.
1. "'3' is not a valid type alias name.'
2. Missing type alias name.

**Actual behavior:**
Errors on neither, and neither alias can be used.

If you have a declaration following the typedef, you will get the name of that declaration instead. In the second case that's OK, but in the first case the '3' should still be an error:

```js
/** @typedef {number} 3 - this should still be an error */ 
var Type1;
/** @typedef {number} */
var Type2; // should be allowed
```

Note that the type name is required: even if you provide a following name to use, we'll treat any valid identifier in the comment text as the type name in preference to the following name:

```js
/** @typedef {number} A type */
var Type3;
```

This defines a type `A` not `Type3`. That's why we should issue an error when, for example, the text is '3' instead of 'A'.
